3rdparty/libcxxrt/CMakeLists.txt

```diff
@@ -1,16 +1,26 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
-# Compile CXXRT
+# Compile libcxx for Open Enclave. Some of these build rules emulate
+# those found in the original sources, but instead this builds an
+# OBJECT library in our CMake graph.
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-tautological-compare")
+add_library(libcxxrt OBJECT
+  libcxxrt/src/dynamic_cast.cc
+  libcxxrt/src/exception.cc
+  libcxxrt/src/guard.cc
+  libcxxrt/src/stdexcept.cc
+  libcxxrt/src/typeinfo.cc
+  libcxxrt/src/memory.cc
+  libcxxrt/src/auxhelper.cc
+  libcxxrt/src/libelftc_dem_gnu3.c)
 
-# build cxxrt sources passing additional options
-add_subdirectory(libcxxrt/src)
-target_compile_options(cxxrt-static PRIVATE
-    -fPIC
-    # cannot use add_link_libraries on out-of-dir target
-    -nostdinc $<$<COMPILE_LANGUAGE:CXX>:-nostdinc++>
-    -I${OE_INCDIR}/openenclave/libc
-    )
-add_dependencies(cxxrt-static oelibc_includes)
+set_property(SOURCE libcxxrt/src/exception.cc PROPERTY COMPILE_FLAGS -Wno-unused-variable)
```

Comment:
Why not set_source_files_properties? (The rationale is not obvious for non CMake experts and therefore I would prefer rationale to be commented)

Reply:
There is honestly no technical difference besides the latter being a bit more specific. But the end result is the same. I tend to prefer more specific commands, so I will change this (I honestly don't remember writing it, it must have been from a stash of a stash of a stash).